### PR TITLE
Refactor LOS extension feature; keep link status

### PR DESCRIPTION
**Pull request by wuweicaifb**
## Prerequisites

- [x] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [x] If this is a non-trivial change, I have already opened an accompanying Issue.
- [x] If applicable, I have included documentation updates alongside my code changes.

## Description

1. Rename feature "Extend Existing Candidate Graph" as "Extend Base Topology" and update its description.
2. Add assert that `building_outline_file_path` and `base_topology_file_path` cannot be set in a plan; Update tests and template to pass tests. Only call `pick_best_sites_per_building` and `select_additional_dns` when site detection is enabled.
3. Keep site and link status in the base topology (we used to only keep site status)
a. Rename `get_all_sites_links_and_exclusion_zones` as `get_los_topology`. This function does not return existing links any more. Instead, it returns a base topology (can be empty topology when base_topology is not provided). 
b. Add a new arg `infer_input_site_location` to function `extract_topology_from_file`, which enables the function to infer altitude and location type for base topology. Update the use of this function in the test code.

## Test Plan

1. Workflows
2. Ran benchmarks 
a. with base topology
b. with building data
c. without either of above

### CODE REVIEW

**Comment by dbeylkin** [comment]
The codecov check fails for some reason. Is this PR still ready for review?

**Comment by wuweicaifb** [comment]
> The codecov check fails for some reason. Is this PR still ready for review?

Yes. Ready for review. I think the way we fix it in #6 is not exactly right. Will look into it later.

**Comment by dbeylkin** [comment]
> > The codecov check fails for some reason. Is this PR still ready for review?
> 
> 
> 
> Yes. Ready for review. I think the way we fix it in #6 is not exactly right. Will look into it later.

I'm a little unclear how these things are related.

1. Why do you think that there's an issue in #6? Is it related to this PR somehow?

2. Codecov seems to complain about uncovered lines in core.py?

3. I'm surprised that the workflow fails. I thought we set up informational only?

**Comment by wuweicaifb** [comment]
> > > The codecov check fails for some reason. Is this PR still ready for review?
> > 
> > 
> > Yes. Ready for review. I think the way we fix it in #6 is not exactly right. Will look into it later.
> 
> I'm a little unclear how these things are related.
> 
> 1. Why do you think that there's an issue in [LOS Rectangle #6](https://github.com/terragraph/terragraph-planner/pull/6)? Is it related to this PR somehow?
> 2. Codecov seems to complain about uncovered lines in core.py?
> 3. I'm surprised that the workflow fails. I thought we set up informational only?

My bad. I should have given more specific description about "the way we fix it in #6". I meant the codecov config change in that PR, rather than the LOS code. I think we need to set something other than "informational" to suppress the workflow failure due to coverage. I will check the codecov doc later again.

File: terragraph_planner/common/configuration/configs.py [hunk 1]
```diff
@@ -399,6 +399,13 @@ def __init__(
                 {"kml", "kmz"},
                 "BASE_TOPOLOGY_FILE_PATH",
             )
+        planner_assert(
+            building_outline_file_path is None
+            or base_topology_file_path is None,
+            "Building outline file and base topology file cannot be both "
```
Comment (dbeylkin) [inline terragraph_planner/common/configuration/configs.py]: Maybe: "Site detection cannot be enabled when a base topology is provided"

File: docs/runbook/Features.md [hunk 2]
```diff
@@ -246,10 +246,13 @@ POPs to the network or reducing the requested demand is a better alternative.
 To use the Maximize Common Bandwidth feature, set `MAXIMIZE_COMMON_BANDWIDTH` under
 the `NETWORK_DESIGN` section to True.
 
-## Extend Existing Candidate Graph
+## Extend Base Topology
 
-This feature enables extending an already deployed "base network" with new
-sites. LOS will be computed between the base network sites and the new sites
+This feature enables extending a base topology with new sites, where a base
```
Comment (dbeylkin) [inline docs/runbook/Features.md]: Maybe: This feature enables the addition of new sites to a base topology with known sites and links. The sites...

LOS will be computed between the sites in the base topology and the new sites and between each of the new sites. The new candidate topology includes the base topology as a subgraph. The candidate topology is then optimized to generate a network plan.

**Review by dbeylkin — APPROVED** [review]

**Pull request merged**